zkevm-circuits/src/evm_circuit/execution/create.rs

```diff
@@ -149,34 +159,34 @@ impl<F: Field, const IS_CREATE2: bool, const S: ExecutionState> ExecutionGadget<
             1.expr(),
         );
 
-        cb.stack_pop(value.expr());
-        cb.stack_pop(init_code.offset_rlc());
-        cb.stack_pop(init_code.length_rlc());
+        cb.stack_pop(value.to_word());
+        cb.stack_pop(init_code.offset_word());
+        cb.stack_pop(init_code.length_word());
         if IS_CREATE2 {
-            cb.stack_pop(create.salt_word_rlc(cb));
+            cb.stack_pop(create.salt());
         }
 
-        cb.stack_push(is_success.expr() * contract_addr_rlc);
+        cb.stack_push(contract_addr.to_word().mul_selector(is_success.expr()));
 
         let (init_code_rlc, keccak_code_hash) = cb.condition(init_code.has_length(), |cb| {
             // the init code is being copied from memory to bytecode, so a copy table lookup to
             // verify that the associated fields for the copy event.
-            let keccak_code_hash = cb.query_cell_phase2();
+            let keccak_code_hash = cb.query_word32();
             let init_code_rlc = cb.query_cell_phase2();
 
             (init_code_rlc, keccak_code_hash)
         });
         cb.condition(not::expr(init_code.has_length()), |cb| {
-            cb.require_equal(
+            cb.require_equal_word(
                 "keccak hash of empty bytes",
-                keccak_code_hash.expr(),
-                cb.empty_keccak_hash_rlc(),
-            );
-            cb.require_equal(
-                "code hash of empty bytes",
-                create.code_hash_word_rlc(),
-                cb.empty_code_hash_rlc(),
+                keccak_code_hash.to_word(),
+                cb.empty_keccak_hash(),
             );
+            // cb.require_equal(
+            //     "code hash of empty bytes",
+            //     create.code_hash(),
+            //     cb.empty_code_hash(),
+            // );
```

Comment:
this is needed.   Add TODO?

Reply:
reenabled in [453ff8a](https://github.com/scroll-tech/zkevm-circuits/pull/963/commits/453ff8a7dd4845c2e4e117b914b03c23f866185a)